lib/src/channels_bloc.dart

```diff
@@ -97,19 +97,19 @@ class ChannelsBlocState extends State<ChannelsBloc>
         paginationParams: paginationParams,
         onlyOffline: onlyOffline,
       )
-          .listen((channels) {
+          .then((channels) {
         if (clear) {
           _channelsController.add(channels);
         } else {
           final l = oldChannels + channels;
           _channelsController.add(l);
         }
-      }, onDone: () {
-        _queryChannelsLoadingController.sink.add(false);
-      }, onError: (err, stackTrace) {
+      }).catchError((err, stackTrace) {
         print(err);
         print(stackTrace);
         _queryChannelsLoadingController.addError(err, stackTrace);
+      }).whenComplete(() {
+        _queryChannelsLoadingController.sink.add(false);
```

Comment:
I think we should set the loading to false in `then` itself as `whenComplete` will get called even after `catchError`.
https://dart.dev/guides/libraries/futures-error-handling#async-try-catch-finally-using-whencomplete